strawberryfields/backends/bosonicbackend/bosoniccircuit.py

```diff
@@ -301,23 +301,29 @@ def fidelity_coherent(self, alpha, modes=None):
         """ Returns a function that evaluates the Q function of the given state """
         if modes is None:
             modes = list(range(self.nlen))
-
-        Q = self.qmat(modes)
-        Qi = np.linalg.inv(Q)
-        delta = self.mean[modes] - alpha
-
-        delta = np.concatenate((delta, np.conjugate(delta)))
-        return np.sqrt(np.linalg.det(Qi).real) * np.exp(
-            -0.5 * np.dot(delta, np.dot(Qi, np.conjugate(delta))).real
+        mode_ind = np.append(2 * np.array(modes), 2 * np.array(modes) + 1)
```

Comment:
I am a bit confused about the quadrature or mode ordering here.

Reply:
Did you try to make a two mode test?

Reply:
I did. The way mode_ind is constructed, it does ends up changing the mode ordering to (q1,...,qn,p1,...,pn) when you call an object like self.means[:,mode_ind], but then this is compensated by the fact that the alpha_mean variable has (q1,...,qn,p1,...,pn) ordering.

Reply:
But then, don't you need to also reorder the covariance matrix? Maybe that is what you do below. In any case I'd suggest you add a comment for future readers of your code so that they are not lost.